### PR TITLE
Adds a new Test "Java Exercises" to garage.com.

### PR DESCRIPTION
**Pull request by renekounkou**
Please review.

### CODE REVIEW

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 1]
```diff
@@ -1,178 +1,125 @@
...
 
 public class JavaExercisesTest {
   @Test
   public void comparaTwoValue() {
-	
 	  int price1 = 45;
 	  int price2 = 45;
-	  
 	  boolean compare = price1==price2;
 	  boolean expectedResult = true;
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java:15]: Move this to the Arrange section (the top of the method)

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 2]
```diff
@@ -1,178 +1,125 @@
...
 	  System.out.println(compare);
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java]: Remove this

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 3]
```diff
@@ -1,178 +1,125 @@
 package com.garage;
-
 import static org.testng.Assert.assertEquals;
 
 import org.testng.annotations.Test;
 
 public class JavaExercisesTest {
   @Test
   public void comparaTwoValue() {
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java]: Fix spelling

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 4]
```diff
@@ -1,178 +1,125 @@
...
   }
-  
-	
   @Test
   public void compareTwoReference() {
 	  String lastName = new String("Kounkou");
 	  String familyName = new String("Kounkou");
-	  
 	  boolean result = lastName == familyName;
 	  boolean expectedResult = false;
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java:25]: Move this to the Arrange section

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 5]
```diff
@@ -1,178 +1,125 @@
...
 	  System.out.println(result);
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java]: Remove this.

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 6]
```diff
@@ -1,178 +1,125 @@
...
   public void demonstrateLogicalANDoperator() {
 	  int age1 = 30;
 	  int age2 = 45;
 	  int totalAge; 
-	  
 	  if(age1==age2 && age2==age1) {
 		  totalAge = 75;
-	  
 	  int expectedResult = 75;
 	  assertEquals(totalAge, expectedResult, "total age is equal to 75");
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java:49]: Assertion should be outside of the IF statement block. The assertion may not be reached due to the way it is currently written.

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 7]
```diff
@@ -1,178 +1,125 @@
...
 	  int age1 = 30;
 	  int age2 = 45;
 	  int totalAge = 75; 
-	  
 	  if(age1==age2 || age2==age1) {
-		  totalAge = 75;
-	  
+	  totalAge = 75;
 	  int expectedResult = 75;
 	  assertEquals(totalAge, expectedResult, "total age is equal to 75");
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java:60]: Same as above.

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 8]
```diff
@@ -1,178 +1,125 @@
...
-	  
 	  }
 }
-  
   @Test
   public void demonstrateUnaryOperstors() {
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java]: Check spelling.

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 9]
```diff
@@ -1,178 +1,125 @@
...
 	  int x = 3;
 	  int y = x++;
 	  int z = x + y;
 	  int w = --z-1;
-	  
 	  System.out.println(z);
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java]: Remove println's

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 10]
```diff
@@ -1,178 +1,125 @@
...
+	  totalAge = 75;
 	  int expectedResult = 75;
 	  assertEquals(totalAge, expectedResult, "total age is equal to 75");
-	  
 	  }
 }
-  
   @Test
   public void demonstrateUnaryOperstors() {
 	  int x = 3;
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java:64]: Simplify this a bit. It's difficult to understand.

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 11]
```diff
@@ -1,178 +1,125 @@
...
+	  assertEquals(w, expectedResult2, "The variable w as a value of 5"); 
   }
-  
   @Test
   public void changeaBooleanVariable() {
 	  int day1 = 7;
 	  int day2 = 7;
-	  
 	  boolean result = day1 == day2;
 	  System.out.println(result);
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java]: Remove println's

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 12]
```diff
@@ -1,178 +1,125 @@
...
-	  
 	  if(afternoonHour == 14) {
-		  System.out.println("Good Afternoon");
-		  
+	  System.out.println("Good Afternoon");  
 	  boolean compareHour = morningHour != afternoonHour;
 	  boolean expectedResult = true;
 	  System.out.println(compareHour);
-	  
 	  assertEquals(compareHour, expectedResult, "The Morning hour is different to the Afternoon hour");
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java]: This does demonstrate if statements but the assertion should assert that the if statements were executed.

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 13]
```diff
@@ -1,178 +1,125 @@
...
-	  
 	  if(passGrade >= 80) {
 		  System.out.println("Good job. You passed. ");
 		  
 	  } else {
 		  System.out.println("You failed.");
-		  
 		  int expectedResult = passGrade;
-		  
 		  assertEquals(passGrade, expectedResult, "If grade is eual to 80, you passed");
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java]: This does demonstrate if statements but the assertion should assert that the if statements were executed.

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 14]
```diff
@@ -1,178 +1,125 @@
...
 			  System.out.println(i);
 			  System.out.println(i1);
-			  
 			  int[] expectedResult = myArray;
-			  
 			  assertEquals(myArray, expectedResult);
-			  
 		  }
 	  }
-
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java:167]: Needs an assertion

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 15]
```diff
@@ -181,20 +128,15 @@ public void demonstrateaTryCatchBlock() {
 	  
 	  }catch (ArithmeticException e) {
 		  System.out.println("Done");
-		 
 	  }
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java:138]: Needs an assertion

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java [hunk 16]
```diff
@@ -181,20 +128,15 @@ public void demonstrateaTryCatchBlock() {
...
   @Test
   public void demonstrateaForeachLoop() {
 	  String[] colors = {"red", "blue", "yellow", "purple", "green", "black"};
-	  
 	  for(String myColors: colors) {
 		  System.out.println(myColors);
 	  }
-	  
 	  String[] expectedResult = colors;
 	  assertEquals(colors, expectedResult);
```
Comment (dentaylor) [inline garage.com/src/test/java/com/garage/JavaExercisesTest.java:149]: This assertion will fail.

**Review by dentaylor — CHANGES_REQUESTED** [review]
Resolve comments. All tests should pass. Add assertions where missing.

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 1 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java:15]: Done

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 2 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java]: Done

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 3 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java]: Done

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 4 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java:25]: Done

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 5 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java]: Done

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 6 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java:49]: Done

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 7 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java:60]: Resolved

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 8 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java]: Resolved

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 9 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java]: Resolved

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 10 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java:64]: Resolved

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 11 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java]: Resolved

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 12 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java]: Resolved

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 13 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java]: Resolved

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 15 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java:138]: I wrote the assertion inside the try block. Do I have to write the assertion in the catch block?

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 16 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java:149]: The test passed after running it.

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 15 above)
Comment (dentaylor) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java:138]: Write it outside of the try/catch block

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 14 above)
Comment (dentaylor) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java:167]: Move the assertion outside of the for loops

**Review by dentaylor — CHANGES_REQUESTED** [review]
See latest comments

File: garage.com/src/test/java/com/garage/JavaExercisesTest.java (hunk 15 above)
Comment (renekounkou) [reply garage.com/src/test/java/com/garage/JavaExercisesTest.java:138]: Resolved

**Comment by renekounkou** [comment]
The assertion has been moved outside of the for loops.

**Review by dentaylor — APPROVED** [review]

**Pull request merged**